framework/meta/src/cargo_toml_contents.rs

```diff
@@ -55,6 +83,88 @@ impl CargoTomlContents {
             .to_string()
     }
 
+    pub fn package_edition(&self) -> String {
+        self.toml_value
+            .get("package")
+            .expect("missing package in Cargo.toml")
+            .get("edition")
+            .expect("missing package name in Cargo.toml")
+            .as_str()
+            .expect("package name not a string value")
+            .to_string()
+    }
+
+    pub fn mvx_dependency_version(&self) -> String {
+        let toml_value = self.dependency("multiversx-sc").unwrap().to_owned();
+
+        if let Value::Table(table) = toml_value {
+            if let Some(version) = table.get("version") {
+                return version.to_string().replace('\"', "");
+            }
+        }
+        panic!("could not find multiversx-sc dependency version in cargo toml")
+    }
+
+    pub fn mvx_dependency_path(&self) -> Option<String> {
+        let toml_value = self.dependency("multiversx-sc").unwrap().to_owned();
+
+        if let Value::Table(table) = toml_value {
+            if let Some(path) = table.get("path") {
+                return Option::Some(path.to_string().replace('\"', ""));
+            }
+        }
+
+        Option::None
+    }
+
+    pub fn get_adapter_dependencies(&self) -> (String, Option<String>) {
+        (self.mvx_dependency_version(), self.mvx_dependency_path())
+    }
+
+    pub fn add_deps(
```

Comment:
`add_wasm_crate_deps`

Reply:
I would make this a private function and move this to a separate file, together with the `generate_wasm_cargo_toml` function.

They are not useful for a general developer who wants to manipulate an arbitrary cargo.toml they are too specific.